src/process_utility.c

```diff
@@ -752,8 +789,13 @@ process_vacuum(ProcessUtilityArgs *args)
 #endif
 			stmt,
 			is_toplevel);
+		foreach (lc, ctx.chunk_pairs)
+		{
+			chunk_pair *cp = (chunk_pair *) lfirst(lc);
+			ts_cm_functions->update_compressed_chunk_relstats(cp->uncompressed_relid,
+															  cp->compressed_relid);
```

Comment:
Since compression chunks cannot be updated, is it necessary to update compressed stat? Or is the name of the function misleading?

Reply:
The purpose of calling analyze on compression hypertable is to update the relstats on that table. This would be called only if someone wants to explicitly do so (  e.g. upgrade to 2.0, we don't have stats on the compressed chunks. Now call analyze <_timescaledb_internal.compression_table> . This will update stats . This is the primary use case for this fix).

Reply:
more details about this in compression.c file changes.

Reply:
Thank you for clarifying!